### PR TITLE
Add JIRA workflow doc

### PR DESCRIPTION
**Pull request by skoczko**

### CODE REVIEW

**Review by zburke — APPROVED** [review]

**Review by craigmcnally — COMMENTED** [review]
I think what you have here is fine.  Did we want to update the process document to include the specific statuses, e.g. on lines 52, 61, 67, 72?

File: TECH_EVAL_JIRA.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Technical Evaluation Process JIRA Workflow
```
Comment (marcjohnson-kint) [inline TECH_EVAL_JIRA.md:1]: Might we want to make it explicit that this is a _module_ technical evaluation, or is `technical evaluation` a shorthand for this?

File: TECH_EVAL_JIRA.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+* The only available issue type and one that should be used for submissions is "Task".
+
+## JIRA Workflow
+![Techical Council Requests Workflow](assets/workflow.png)
+
+**SUBMITTED** Initial status for newly created submissions.
+
+**UNDER EVALUATION** Once the sumbmission is reviewed by the TC and an evaluation group is formed, the TC indicates this by transitioning the issue to this state.
+
+**CANCELLED** If the submitter decides to withdraw the submission (alone or together with the evalation group), he/she transitions the issue to this state.
```
Comment (marcjohnson-kint) [inline TECH_EVAL_JIRA.md:17]:
```suggestion
**CANCELLED** If the submitter decides to withdraw the submission (alone or together with the evalation group), they transition the issue to this state.
```

File: NEW_MODULE_TECH_EVAL.MD [hunk 3]
```diff
@@ -9,7 +9,7 @@ This document outlines the processes for **technical** evaluation of module for
 1. Approval/Rejection
 
 ### Supporting documentation
-* Details of the JIRA workflow are captured in a separate document:  (insert link). 
+* Details of the JIRA workflow are captured in a separate [document](TECH_EVAL_JIRA.md).
```
Comment (marcjohnson-kint) [inline NEW_MODULE_TECH_EVAL.MD:12]: I think the path to that has now changed.

```suggestion
* Details of the JIRA workflow are captured in a separate [document](NEW_MOD_TECH_EVAL_JIRA.md). 
```

File: NEW_MODULE_TECH_EVAL.MD [hunk 4]
```diff
@@ -45,11 +45,12 @@ Information gathered includes:
 ## Evaluation
 1. The Technical Council reviews a JIRA board to see if there are any new submissions, and check on the status of evaluations in progress.
 1. If there’s a new submission, the Technical Council finds one or more people to do the evaluation. The evaluation team will be formed in a way to allow for a fair review and avoid conflicts for interests.
-1. The module is evaluated.
+    * JIRA workflow: the ticket is transitioned from **SUBMITTED** to **UNDER EVALUATION**
+3. The module is evaluated.
```
Comment (craigmcnally) [inline NEW_MODULE_TECH_EVAL.MD:49]:
```suggestion
1. The module is evaluated.
```

File: NEW_MODULE_TECH_EVAL.MD [hunk 5]
```diff
@@ -45,11 +45,12 @@ Information gathered includes:
...
     * If the evaluator(s) have questions, they may reach out to the provided points of contact.
     * If answers aren't provided in a timely manner, the criteria in question may be failed.
-1. Upon completion of the evaluation the JIRA will be updated to indicate this.
-    * See JIRA workflow for details.
+4. Upon completion of the evaluation the JIRA will be updated to indicate this.
```
Comment (craigmcnally) [inline NEW_MODULE_TECH_EVAL.MD:52]:
```suggestion
1. Upon completion of the evaluation the JIRA will be updated to indicate this.
```

File: NEW_MODULE_TECH_EVAL.MD [hunk 6]
```diff
@@ -58,15 +59,15 @@ Information gathered includes:
...
     * An [Evaluation Results Template](insert link when available) will be used for this.
 1. Interested parties (e.g. release coordinator, contributor points of contact, Product Council members, etc.) are notified, and provided a link to the results.
-    * See JIRA workflow for details.
+    * JIRA workflow: notification is in the form of transition from **TC REVIEW** to **APROVED** or **REJECTED**
     * If failed, the contributors may ask for help understanding the failed criteria.
       * The comments section of the JIRA may be used for this purpose.  
       * If a meeting is required or desired, it's the responsibility of the contributor(s) to find a time that works and set the meeting up with the review team.  The regular Technical Council meeting ***will not*** be used for this purpose.
 1. If applicable, the contributor(s) resolve all issues and request a re-review.  
-    * See JIRA workflow for details.
+    * * JIRA workflow: the ticket is transitioned from  **REJECTED** to **SUBMITTED**
```
Comment (craigmcnally) [inline NEW_MODULE_TECH_EVAL.MD:73]:
```suggestion
    * JIRA workflow: the ticket is transitioned from  **REJECTED** to **SUBMITTED**
```

**Review by craigmcnally — APPROVED** [review]
a few minor syntax comments, but overall LGTM

**Pull request merged**